Add How-To Guides (#15)
pace-neutrons/resolution_functions: docs/source/howtos/optimise_experiment.rst

:term:`instrument` properties and check whether the other choices (e.g.
:term:`configuration`) made seem about right. If everything checks out, the next
step is to plot the resolution at different combinations of the
:term:`settings<setting>`:

>>> data.restrictions
{'e_init': [0, 2000], 'chopper_frequency': [50, 601, 50]}

The ``restrictions`` provide information about the allowed values for each
:term:`setting`. In the case of the MAPS instrument, there are two different
settings:

* ``e_init`` for which all values between ``0`` and ``2000`` meV are allowed
* ``chopper_frequency`` for which all values between ``50`` and ``600``
  (inclusive) at an interval of ``50`` are allowed.

However, depending on the instrument and the amount of scientific knowledge
about the system, these values can be further constrained. For example, MAPS is
a direct-geometry instrument, which means that only the values of energy
transfer up to the value of ``e_init`` (the incident energy) are observable.
Therefore, if it is known that in a given system features between 100 and 600
meV are of interest, the ``e_init`` setting can be restricted further to values
above 600 meV:

>>> max_feature = 600

> >>> test_e_init = np.arange(max_feature, data.restrictions['e_init'][0], 100)

Reviewer: The plot looks right, but I'm surprised that `data.restrictions['e_init'][0]` would be the _upper_ energy limit? Is that correct? Either way, perhaps `max(data.restrictions['e_init']` would be clearer.

Author: no, you're right, that is a mistake. `max` seems reasonable

>>> test_choppers = np.arange(*data.restrictions['chopper_frequency'])

All the data can then be generated using by loopiing over these variables:

>>> energy_transfer = np.arange(0, max_feature, 5)
>>> results = np.zeros((len(test_choppers), len(test_e_init), len(energy_transfer)))
>>> for i, chopper_frequency in enumerate(test_choppers):
...     for j, e_init in enumerate(test_e_init):
...         model = maps.get_resolution_function('PyChop_fit', chopper_package='B', e_init=e_init, chopper_frequency=chopper_frequency)
...         results[i, j, :] = model(energy_transfer)

This can then be plotted in various ways. For example, we can check the effect
of ``e_init`` by plotting the resolution for its different values at a constant
``chopper_frequency``, e.g. the default of 400 Hz:

.. image:: ./optimise_experiment_pictures/maps_b_400.png

from which we might conclude that a low value for ``e_init`` is desirable, so we
might follow up by comparing the different values of ``chopper_frequency`` at
the lowest value of ``e_init``, 600:

.. image:: ./optimise_experiment_pictures/maps_b_e600.png

> which might suggest that low values of ``chopper_frequency`` are ideal. Then, if

Reviewer: The low values have huge broadening, though? Perhaps something like

> which might suggest that ``chopper_frequency`` of 300 Hz and above gives tolerable resolution.

I think in practice resolution is not the only factor determining chopper frequency (or it would always run at the highest setting!) Perhaps @mducle can help make this story sensible. (But we don't need to wait for those improvements before merge.)

the features are expected to be further apart than the resolution, they should
be distinguishable in an INS experiment. If that is not the case, it might mean
that this combination of :term:`instrument`,
:term:`configurations<configuration>`, and :term:`settings<setting>` may not be
suitable for the given system.

> Using computation spectra

Reviewer: ```suggestion
Using simulated spectra
```

-------------------------

The other way to estimate the optimal experimental :term:`settings<setting>` is

> to use real computational spectra, e.g. from

Reviewer: ```suggestion
to simulate spectra from *ab initio* lattice dynamics or molecular dynamics, e.g. using
```

`AbINS <https://github.com/mantidproject/mantid/tree/main/scripts/abins>`_ or
`dynasor <https://dynasor.materialsmodeling.org/index.html>`_, and then convolve
the ResINS resolution using a library like

> `euphonic <https://euphonic.readthedocs.io/en/stable/>`_. Similarly to above,
> though, to do this, the :term:`instrument`, the :term:`model`, and the
> :term:`option` for each :term:`configuration` have to have been chosen. The
> process also starts the same by creating the instrument:

Reviewer: ```suggestion
`euphonic <https://euphonic.readthedocs.io/en/stable/>`_.
Again, we begin by creating the Instrument:
```
